controllers/user-routes.js

```diff
@@ -52,19 +52,41 @@ router.get('/newPost', async (req, res) => {
         if (user) {
            res
             .status(200)
-            .render('new-post', { user }); 
+            .render('add-post') 
         }
 
-        if(!user){
+        else if(!user){
             res
-            .status(400)
-            .render('new-post', { message: 'Please login or register to make a new post.'});
+            .status(200)
+            .render('add-post', { message: 'Please login or register to make a new post.'})
         }
     } catch (err) {
       console.log(err);
         res.status(500).json(err);
     }
   });
```

Comment:
It's better practice to check for the negative status first, also "else" already counts for the inverse of "user", so another "if" is not necessary.
`        if (!user) {
            res
            .status(200)
            .render('add-post', { message: 'Please login or register to make a new post.'})
        } else {
           res
            .status(200)
            .render('add-post') 
        }
    } catch (err) {
      console.log(err);
        res.status(500).json(err);
    }`